### ISSUE TITLE
install fails on node v12

### ISSUE THREAD

**Issue by cainvommars** [joyent/node-bunyan-syslog#53]
Anyone else have the issue that it can't be installed on latest nodejs (and npm) ?

 ~/git/tmp/bunyaaan # node --version && npm --version
v12.4.0
6.9.0
 ~/git/tmp/bunyaaan # npm install bunyan-syslog

> bunyan-syslog@0.3.2 install /Users/peter/git/tmp/bunyaaan/node_modules/bunyan-syslog
> node-gyp rebuild

  CXX(target) Release/obj.target/syslog/src/syslog.o
../src/syslog.cc:60:2: error: no matching member function for call to 'ToString'
        REQUIRE_STRING_ARG(info, 0, ident);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:38:33: note: expanded from macro 'REQUIRE_STRING_ARG'
        String::Utf8Value VAR(ARGS[I]->ToString());
                              ~~~~~~~~~^~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2528:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                           ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2544:35: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<String> ToString(Isolate* isolate) const);
                                  ^
../src/syslog.cc:61:2: error: no matching member function for call to 'ToInteger'
        REQUIRE_INT_ARG(info, 1, logopt);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:31:35: note: expanded from macro 'REQUIRE_INT_ARG'
        Local<Integer> _ ## VAR(ARGS[I]->ToInteger());                  \
                                ~~~~~~~~~^~~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2534:45: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Integer> ToInteger(
                                            ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2548:36: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<Integer> ToInteger(Isolate* isolate) const);
                                   ^
../src/syslog.cc:62:2: error: no matching member function for call to 'ToInteger'
        REQUIRE_INT_ARG(info, 2, facility);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:31:35: note: expanded from macro 'REQUIRE_INT_ARG'
        Local<Integer> _ ## VAR(ARGS[I]->ToInteger());                  \
                                ~~~~~~~~~^~~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2534:45: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Integer> ToInteger(
                                            ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2548:36: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<Integer> ToInteger(Isolate* isolate) const);
                                   ^
../src/syslog.cc:72:2: error: no matching member function for call to 'ToInteger'
        REQUIRE_INT_ARG(info, 0, priority);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:31:35: note: expanded from macro 'REQUIRE_INT_ARG'
        Local<Integer> _ ## VAR(ARGS[I]->ToInteger());                  \
                                ~~~~~~~~~^~~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2534:45: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Integer> ToInteger(
                                            ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2548:36: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<Integer> ToInteger(Isolate* isolate) const);
                                   ^
../src/syslog.cc:73:2: error: no matching member function for call to 'ToString'
        REQUIRE_STRING_ARG(info, 1, message);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:38:33: note: expanded from macro 'REQUIRE_STRING_ARG'
        String::Utf8Value VAR(ARGS[I]->ToString());
                              ~~~~~~~~~^~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2528:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                           ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2544:35: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<String> ToString(Isolate* isolate) const);
                                  ^
../src/syslog.cc:91:2: error: no matching member function for call to 'ToInteger'
        REQUIRE_INT_ARG(info, 0, maskpri);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/syslog.cc:31:35: note: expanded from macro 'REQUIRE_INT_ARG'
        Local<Integer> _ ## VAR(ARGS[I]->ToInteger());                  \
                                ~~~~~~~~~^~~~~~~~~
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2534:45: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Integer> ToInteger(
                                            ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:2548:36: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<Integer> ToInteger(Isolate* isolate) const);
                                   ^
../src/syslog.cc:98:6: error: variable has incomplete type 'void'
void init(Handle<Object> target) {
     ^
../src/syslog.cc:98:11: error: use of undeclared identifier 'Handle'
void init(Handle<Object> target) {
          ^
../src/syslog.cc:98:18: error: 'Object' does not refer to a value
void init(Handle<Object> target) {
                 ^
/Users/peter/.node-gyp/12.4.0/include/node/v8.h:3356:17: note: declared here
class V8_EXPORT Object : public Value {
                ^
../src/syslog.cc:98:26: error: use of undeclared identifier 'target'
void init(Handle<Object> target) {
                         ^
../src/syslog.cc:98:33: error: expected ';' after top level declarator
void init(Handle<Object> target) {
                                ^
                                ;
11 errors generated.
make: *** [Release/obj.target/syslog/src/syslog.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/peter/.nvm/versions/node/v12.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/Users/peter/.nvm/versions/node/v12.4.0/bin/node" "/Users/peter/.nvm/versions/node/v12.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/peter/git/tmp/bunyaaan/node_modules/bunyan-syslog
gyp ERR! node -v v12.4.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN bunyaaan@1.0.0 No description
npm WARN bunyaaan@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bunyan-syslog@0.3.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bunyan-syslog@0.3.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by cainvommars** [comment]
are there any plans to invest in this? @cburroughs (just picking you because you did the last commits on master)

**Comment by melloc** [comment]
Fixed in 1261cf7.
